src/video.cpp

```diff
@@ -465,7 +457,11 @@ void VideoSystem::MainLoop()
 		for (;;) {
 			if (HandleEvent()) break;
 		}
-		if (_finish) break;
+		/* If necessary, change the played game. */
+		if (_game_control.next_action != GCA_NONE) {
+			_game_control.RunAction();
+		}
+		if (!_game_control.running) break;
```

Comment:
Could make it an inline method, not sure it's worth the effort for 1 boolean test.